src/TokenManagers/XboxTokenManager.js

```diff
@@ -24,50 +24,26 @@ class XboxTokenManager {
     this.headers = { 'Cache-Control': 'no-store, must-revalidate, no-cache', 'x-xbl-contract-version': 1 }
   }
 
-  async getCachedTokens () {
+  async getCachedTokens (relyingParty) {
     const cache = await this.cache.getCached()
     const tokens = {}
-    for (const type of ['userToken', 'titleToken', 'deviceToken']) {
+    for (const type of ['userToken', 'titleToken', 'deviceToken', createHash(relyingParty)]) {
```

Comment:
Can you explain what's going on here and why there needs to be a for loop with a createHash?

Reply:
instead of having, `getCachedUserToken`, `getCachedDeviceToken`, `getCachedTitleToken` and `getCachedXstsToken` we have a single function to handle fetching tokens which means well only make one call to the cache vs 4. We need the `createHash` function to fetch the xsts token which is stored with a hash of the relyingParty.